sklearn/linear_model/tests/test_sgd.py

```diff
@@ -1338,6 +1387,303 @@ def test_loss_function_epsilon(klass):
+# SGD One Class SVM Test Case
+
+# a simple implementation of ASGD to use for testing SGDOneClassSVM
+def asgd_oneclass(klass, X, eta, nu, coef_init=None, offset_init=0.0):
+    if coef_init is None:
+        coef = np.zeros(X.shape[1])
+    else:
+        coef = coef_init
+
+    average_coef = np.zeros(X.shape[1])
+    offset = offset_init
+    intercept = 1 - offset
+    average_intercept = 0.0
+    decay = 1.0
+
+    # sparse data has a fixed decay of .01
+    if klass == SparseSGDOneClassSVM:
+        decay = .01
+
+    for i, entry in enumerate(X):
+        p = np.dot(entry, coef)
+        p += intercept
+        if p <= 1.0:
+            gradient = -1
+        else:
+            gradient = 0
+        coef *= max(0, 1.0 - (eta * nu / 2))
+        coef += -(eta * gradient * entry)
+        intercept += -(eta * (nu + gradient)) * decay
+
+        average_coef *= i
+        average_coef += coef
+        average_coef /= i + 1.0
+
+        average_intercept *= i
+        average_intercept += intercept
+        average_intercept /= i + 1.0
+
+    return average_coef, 1 - average_intercept
+
+
+@pytest.mark.parametrize('klass', [SGDOneClassSVM, SparseSGDOneClassSVM])
+@pytest.mark.parametrize('nu', [-0.5, 2])
+def test_bad_nu_values(klass, nu):
+    msg = r"nu must be in \(0, 1]"
+    assert_raises_regexp(ValueError, msg, klass, nu=nu)
+
+    clf = klass(nu=0.05)
+    clf2 = clone(clf)
+    assert_raises_regexp(ValueError, msg, clf2.set_params, nu=nu)
+
+
+@pytest.mark.parametrize('klass', [SGDOneClassSVM, SparseSGDOneClassSVM])
+def _test_warm_start_oneclass(klass, X, lr):
+    # Test that explicit warm restart...
+    clf = klass(nu=0.5, eta0=0.01, shuffle=False,
+                learning_rate=lr)
+    clf.fit(X)
+
+    clf2 = klass(nu=0.1, eta0=0.01, shuffle=False,
+                 learning_rate=lr)
+    clf2.fit(X, coef_init=clf.coef_.copy(),
+             offset_init=clf.offset_.copy())
+
+    # ... and implicit warm restart are equivalent.
+    clf3 = klass(nu=0.5, eta0=0.01, shuffle=False,
+                 warm_start=True, learning_rate=lr)
+    clf3.fit(X)
+
+    assert clf3.t_ == clf.t_
+    assert_array_almost_equal(clf3.coef_, clf.coef_)
+
+    clf3.set_params(nu=0.1)
+    clf3.fit(X)
+
+    assert clf3.t_ == clf2.t_
+    assert_array_almost_equal(clf3.coef_, clf2.coef_)
+
+
+@pytest.mark.parametrize('klass', [SGDOneClassSVM, SparseSGDOneClassSVM])
+@pytest.mark.parametrize('lr',
+                         ["constant", "optimal", "invscaling", "adaptive"])
+def test_warm_start_oneclass(klass, lr):
+    _test_warm_start_oneclass(klass, X, lr)
+
+
+@pytest.mark.parametrize('klass', [SGDOneClassSVM, SparseSGDOneClassSVM])
+def test_clone_oneclass(klass):
+    # Test whether clone works ok.
+    clf = klass(nu=0.5)
+    clf = clone(clf)
+    clf.set_params(nu=0.1)
+    clf.fit(X)
+
+    clf2 = klass(nu=0.1)
+    clf2.fit(X)
+
+    assert_array_equal(clf.coef_, clf2.coef_)
+
+
+@ignore_warnings
```

Comment:
Can you please add a comment to explain which warning we want to ignore and why is it ok to ignore them in this test?

Please also pass the warning class as the category argument to this decorator, e.g. `@ignore_warnings(category=ConvergenceWarning)`.

Reply:
Actually isn't there an easy way to avoid triggering the warning in this test in the first place?

Reply:
I removed this `ignore_warnings`, apparently no warning is raised.

Reply:
Indeed CIs are still green, thanks for spotting this